### ISSUE TITLE
Retinanet do NOT produce any prediction with MobileNet or DenseNet ?!?

### ISSUE THREAD

**Issue by thusinh1969** [fizyr/keras-retinanet#1012]
One thing I scratch my head for sometime to find how to train Retinanet with other models such as MobileNet224 or DensetNet209 etc. Seems there is hidden bugs somewhere that do not make the training work with these model as no prediction at all after training. ResNet model works very well though.

Have been crazy lately really as my food dataset worked much better with Retinanet ResNet152 than Tensorflow model zoo (Inception or Resnet101). Would really like to train Retinanet for mobile as well.

Any hint please. Thanks.
Steve

**Comment by mrkaworu** [comment]
I couldn't get MobileNet224 or 192 to work but Resnet50, 101 and vgg16 worked for me.

**Comment by thusinh1969** [comment]
> I couldn't get MobileNet224 or 192 to work but Resnet50, 101 and vgg16 worked for me.

Exactly. I thought that lots of people would want Retinanet to work on mobile as Retinanet proven to give much better accuracy at small objects in images with FPN-alike concept vs SSD single-pass alone which missed lot of small objects.

Also I am pretty sure that some people would be happy to sacrify some speed for accuracy as real-time is not a must have on mobile sometime. Retinanet is now state-of-the-art, please make this package works with other backbone likes MobileNet and DenseNet and such as guided in your README.md.

Thanks Fizyr and team.

**Comment by hgaiser** [comment]
Unfortunately our resources are limited. We use frequently use resnet so you could say that it is "officially supported". Other backbones, like mobilenet and densenet are not used so they might have bugs. If you would like to work on it, to find out what the issues are, then that would be appreciated. For now I will close this issue because there is no specific problem, but if you find something wrong then feel free to create a pull request.

**Comment by ikerodl96** [comment]
It uses the method retinanet_bbox(...) instead of retinanet(...) of retinanet.py
